File: 프로그래머스/lv1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.js
function solution(board, moves) {
    let answer = 0;
    let basket = [];

    moves.forEach((e) => {
        for (let i = 0; i < board.length; i++) {
            board[i][e - 1] !== 0 ? (
                basket.at(-1) !== board[i][e - 1] ? 
                    basket.push(board[i][e - 1]) : (basket.splice(basket.length - 1, 1), answer += 2),
                board[i].splice(e - 1, 1, 0), i = board.length
            ) : -1;
        }
    })
    
    return answer;
}

File: 프로그래머스/1/64061. 크레인 인형뽑기 게임/크레인 인형뽑기 게임.java
import java.util.*;

class Solution {
    public int solution(int[][] board, int[] moves) {
        int answer = 0;
        Stack<Integer> s = new Stack<>();
        
        for (int m : moves) {
            for (int i = 0; i < board.length; i++) {
                
                if (board[i][m - 1] != 0) {
                    
                    int target = board[i][m - 1];
                    if (!s.empty() && s.peek() == target) {
                        s.pop();
                        answer++;
                    } else {
                        s.push(board[i][m - 1]);
                    }
                    
                    board[i][m - 1] = 0;
                    break;
                }
            }
        }
        
        return answer * 2;
    }
}